firebase-firestore/ktx/src/main/kotlin/com/google/firebase/firestore/ktx/Firestore.kt

```diff
@@ -163,3 +167,35 @@ class FirebaseFirestoreKtxRegistrar : ComponentRegistrar {
     override fun getComponents(): List<Component<*>> =
             listOf(LibraryVersionComponent.create(LIBRARY_NAME, BuildConfig.VERSION_NAME))
 }
+
+/**
+ * Attach a snapshotListener to a DocumentReference and use it as a coroutine flow
+ */
+fun DocumentReference.toFlow() = callbackFlow {
+    val listener = addSnapshotListener { value, error ->
```

Comment:
We should allow the user to specify the `metadataChanges`parameter:

```kotlin
fun DocumentReference.toFlow(metadataChanges: MetadataChanges = MetadataChanges.EXCLUDE) = callbackFlow {
    val listener = addSnapshotListener(metadataChanges) { value, error ->
```

I'm unsure about the `executor` parameter. This would typically be handled by coroutine Dispatchers but maybe there's a use case ?

Reply:
I have just added the `MetadataChanges` parameter.
Thanks 👍